go/types/struct.go

```diff
@@ -195,13 +194,87 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	return escapeFields(input, encode)
+}
+
+func CamelCaseFieldName(input string) string {
+	encode := func(s1 string, p *regexp.Regexp) string {
+		if p.MatchString(s1) {
+			return s1
+		}
+		return ""
+	}
 
+	splitFields := strings.Fields(input)
+	output := escapeFields(splitFields[0], encode)
+
+	if len(splitFields) > 1 {
+		output = strings.ToLower(output)
+	}
+
+	for _, field := range splitFields[1:] {
+		output += strings.Title(strings.ToLower(escapeFields(field, encode)))
+	}
+
+	if !IsValidStructFieldName(output) {
+		return ""
+	}
+	return output
+}
+
+func escapeFields(input string, encode encodingFunc) string {
 	output := ""
-	pattern := headPattern
+	pattern := headFieldNamePattern
 	for _, ch := range input {
 		output += encode(string([]rune{ch}), pattern)
-		pattern = tailPattern
+		pattern = tailFieldNamePattern
 	}
-
 	return output
 }
+
+// EscapeStructField escapes names for use as noms structs. Disallowed characters are encoded as
```

Comment:
Actually, sorry, comment is correct. But can we just collapse this with QEncodeFieldName? do we need both?